### PR TITLE
Extract _get_op_def_compute_fn into wrap_source_asset_observe_fn_in_op_compute_fn

### PR DESCRIPTION
**Pull request by schrockn**
## Summary & Motivation

This is a simple refactoring so that I can use this function to implement #16620

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by schrockn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16617** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16617" target="_blank">Graphite</a> 
    * **PR #16618** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16618" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16618?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/source_asset.py [hunk 1]
```diff
@@ -50,10 +50,87 @@
 from dagster._utils.merger import merge_dicts
 from dagster._utils.warnings import disable_dagster_warnings
 
+# getting the following error
+# python_modules/dagster/dagster/_core/definitions/source_asset.py:55:9: TCH001 Move application import `dagster._core.definitions.decorators.op_decorator.DecoratedOpFunction` into a type-checking block
+# if TYPE_CHECKING:
+#     from dagster._core.definitions.decorators.op_decorator import (
+#         DecoratedOpFunction,
+#     )
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/source_asset.py:58]: [1]

File: python_modules/dagster/dagster/_core/definitions/source_asset.py [hunk 2]
```diff
@@ -50,10 +50,87 @@
...
+
 # Going with this catch-all for the time-being to permit pythonic resources
 SourceAssetObserveFunction: TypeAlias = Callable[..., Any]
 
 
+@staticmethod
+def wrap_source_asset_observe_fn_in_op_compute_fn(
+    source_asset: "SourceAsset",
+):
+    # ) -> "DecoratedOpFunction":
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/source_asset.py:68]: [2]

File: python_modules/dagster/dagster/_core/definitions/source_asset.py [hunk 3]
```diff
@@ -50,10 +50,87 @@
 from dagster._utils.merger import merge_dicts
 from dagster._utils.warnings import disable_dagster_warnings
 
+# getting the following error
```
[the remark this answers is not in the document]
Comment (alangenfeld) [reply python_modules/dagster/dagster/_core/definitions/source_asset.py:53]: I assume if you uncomment [1] and [2] things should work - ruff just wants you to import in the TYPE_CHECKING block if you only use the type as a typehint like in [2] 

its unclear to me if [1] is part of the error message you got for a import no longer present, or if you had [1] and commented it out and put the error above it

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**